### PR TITLE
Added min version requirements and README instructions for torchdata

### LINKED ISSUE #2047: Update `CONTRIBUTING.md` w/ instruction on how to install `torchdata` from source
See https://github.com/pytorch/text/issues/2045

### PR DESCRIPTION
**Pull request by Nayef211**
Taking suggestions from @ejguan comments in https://github.com/pytorch/text/issues/2045#issuecomment-1416182472

Closes #2047

### CODE REVIEW

**Review by ejguan — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -19,7 +19,7 @@ Sphinx
 pytest
 expecttest
 parameterized
-torchdata
+torchdata>0.5
```
Comment (ejguan) [inline requirements.txt:22]: You might want to capture in your release runbook to update this version per release.

Comment (Nayef211) [reply requirements.txt:22]: Done

**Review by mthrok — APPROVED** [review]

**Comment by rshraga** [comment]
@Nayef211 it looks like after this landed, many nightlies started failing due to `import torchdata` statement in `text/test/smoke_tests/smoke_tests.py` https://github.com/pytorch/text/actions/runs/4113901586/jobs/7100606817

**Comment by ejguan** [comment]
@rshraga Thanks for reporting this. This is expected until the PyTorch nightly has been updated tmrw.

**Pull request merged**